types/events.go

```diff
@@ -137,6 +165,37 @@ func ParseTypedEvent(event abci.Event) (proto.Message, error) {
 	return protoMsg, nil
 }
 
+// ResultEventToABCIEvent takes the ctypes.ResultEvent and casts it to abci.TxResult, extracting the []abci.Event
+func ResultEventToABCIEvent(rev ctypes.ResultEvent) ([]abci.Event, error) {
```

Comment:
Shouldn't this also return some sort of boolean flag indicating whether the returned events are `TxEvents` or `BlockEvents`?